lib/config/flat-config-array.js

```diff
@@ -129,6 +187,49 @@ class FlatConfigArray extends ConfigArray {
         Object.defineProperty(this, "shouldIgnore", { writable: false });
     }
 
+    /**
+     * Normalizes the array by calling the superclass method and catching/rethrowing
+     * any ConfigError exceptions with additional details.
+     * @param {any} [context] The context to use to normalize the array.
+     * @returns {Promise<void>} A promise that resolves when the array is normalized.
+     */
+    normalize(context) {
+        return super.normalize(context)
+            .catch(error => {
+                if (error.name === "ConfigError") {
+                    throw wrapConfigErrorWithDetails(error, this[originalLength], this[baseLength]);
+                }
+
+                throw error;
+
+            });
+    }
+
+    /**
+     * Normalizes the array by calling the superclass method and catching/rethrowing
+     * any ConfigError exceptions with additional details.
+     * @param {any} [context] The context to use to normalize the array.
+     * @returns {void} A promise that resolves when the array is normalized.
```

Comment:
`super.normalizeSync` returns `this`.

```suggestion
     * @returns {FlatConfigArray} The current instance.
```